src/matchers/utils.ts

```diff
@@ -1,21 +1,23 @@
 import type { Page, ElementHandle, Frame } from "playwright-core"
 import { PageWaitForSelectorOptions } from "../../global"
 
-export type ExpectInputType = Page | Frame | ElementHandle
+type Handle = Page | Frame | ElementHandle
+export type ExpectInputType = Handle | Promise<Handle>
 
-const isElementHandle = (value: ExpectInputType): value is ElementHandle => {
+const isElementHandle = (value: Handle): value is ElementHandle => {
   return value.constructor.name === "ElementHandle"
 }
 
 export const getFrame = async (value: ExpectInputType) => {
-  return isElementHandle(value) ? value.contentFrame() : value
+  const resolved = await value
+  return isElementHandle(resolved) ? resolved.contentFrame() : resolved
```

Comment:
`getFrame` is used by some matchers that don't use `getElementHandle` so we need to await the value here as well.